### DISCUSSION TITLE
Are duplicateagents a bad thing...?

### DISCUSSION THREAD

**Discussion by Xevichr** [Ylianst/MeshCentral#6205] [Q&A] [answered]
When running `agentissues` it will show many duplicate agents. What causes this in the first place? 

I am assuming it is caused by deploying the Agent the way I am. We are using Microsoft SCCM to deploy the MeshAgent to machines. It will use a .exe (With a set device group) installer and -fullinstall. All machines will use the same .exe for this.

While I don't quite see any problems so far, because I can see many devices that are supposedly a "duplicateagent" just fine in MeshCentral, won't it cause problems in the long run? What are the downsides? 

Besides that, but that's a whole different story, I am also looking into "Bad Signature" Agents. Its not like that is somehow caused by the way I distribute the Agents to machines right?

I wonder if others are having the same thing regarding the duplicate Agents, because before enrolling the Agent through SCCM I did find people that are all seemingly deploying it the same way, through one specific installer.

https://www.reddit.com/r/MeshCentral/comments/un4jsp/meshagent_deployment/
https://www.reddit.com/r/MeshCentral/comments/ma51bc/installing_windows_10_agent_silent_and_unattended/

The only exception I found is this fellow:

https://www.reddit.com/r/MeshCentral/comments/ma51bc/comment/grr2jky/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

**Comment by si458** [comment, marked as answer]
from my understanding after quickly looking at the code, the duplicategent is simply,

1. the agent device connected to meshcentral 1st time
2. then the connection gets dropped for some reason (comp cpu usage too high, no free memory, no internet, etc...)
3. the agent device then reconnects
4. meshcentral sees that the agent device is STILL connected from the first time around
5. meshcentral then logs this so you was aware
6. meshcentral then disconnects the first connection, and lets the second connection have priority/access instead

so its nothing major to worry about as you might see the odd 1 or 2 duplicate agents for hundreds of devices

BUT if you saw hundreds of duplicate agents then the is an issue somewhere weather its an internet/connectivity issue,
vps issue, or simply your reverse proxy is disconnecting the websocket connections for some reason?

**Reply by Xevichr** [reply]
It seems likely that connectivity is the issue. We have people working from home using a VPN, and when the VPN disconnects, the Agent will too. Currently, MeshCentral is running on LAN only for us.

Thank you for your response. It appears I completely misinterpreted what a duplicate agent might be. In hindsight, it makes sense that it relates to the connection, as a duplicate connection would naturally occur due to an improperly terminated old connection.

In any case thanks again :)
